packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -752,17 +816,6 @@ function getRequestModelMapping(
         )} ${
           !property.optional ? "" : `!${propertyFullName} ? undefined :`
         } ${propertyFullName}`;
-      } else if (
-        (property.restApiName === "message" ||
-          property.restApiName === "messages") &&
-        (property.type.name === "ChatMessage" ||
-          property.type.elementType?.name === "ChatMessage")
-      ) {
-        definition = `"${property.restApiName}": ${
-          !property.optional
-            ? `${propertyFullName} as any`
-            : `!${propertyFullName} ? undefined : ${propertyFullName} as any`
-        }`;
```

Comment:
This is the workaround for recursive reference in codegen.

Reply:
Yeah I wanted to try this out without the workaround, since the properties in message don't need to be reshaped I think we should be fine without it